syncd/syncd.cpp

```diff
@@ -3087,35 +3158,50 @@ sai_status_t processEvent(
 
             internal_syncd_get_send(object_type, str_object_id, switch_vid, status, attr_count, attr_list);
         }
-        else if (status != SAI_STATUS_SUCCESS)
+        else
         {
             internal_syncd_api_send_response(api, status);
 
-            if (!info->isnonobjectid && api == SAI_COMMON_API_SET)
+            notifSent = notifySAIAPIStatus(kco, status);
+
+            if (status != SAI_STATUS_SUCCESS)
             {
-                sai_object_id_t vid;
-                sai_deserialize_object_id(str_object_id, vid);
+                if (!info->isnonobjectid && api == SAI_COMMON_API_SET)
+                {
+                    sai_object_id_t vid;
+                    sai_deserialize_object_id(str_object_id, vid);
 
-                sai_object_id_t rid = translate_vid_to_rid(vid);
+                    sai_object_id_t rid = translate_vid_to_rid(vid);
 
-                SWSS_LOG_ERROR("VID: %s RID: %s",
-                        sai_serialize_object_id(vid).c_str(),
-                        sai_serialize_object_id(rid).c_str());
-            }
+                    SWSS_LOG_ERROR("VID: %s RID: %s",
+                            sai_serialize_object_id(vid).c_str(),
+                            sai_serialize_object_id(rid).c_str());
+                }
 
-            for (const auto &v: values)
+                for (const auto &v: values)
+                {
+                    SWSS_LOG_ERROR("attr: %s: %s", fvField(v).c_str(), fvValue(v).c_str());
+                }
+
+                if (notifSent)
+                {
+                    SWSS_LOG_ERROR("failed to execute api: %s, key: %s, status: %s",
+                            op.c_str(),
+                            key.c_str(),
+                            sai_serialize_status(status).c_str());
+                }
+                else
+                {
+                    SWSS_LOG_THROW("failed to execute api: %s, key: %s, status: %s",
+                            op.c_str(),
+                            key.c_str(),
+                            sai_serialize_status(status).c_str());
+                }
+            }
+            else // non GET api, status is SUCCESS
             {
-                SWSS_LOG_ERROR("attr: %s: %s", fvField(v).c_str(), fvValue(v).c_str());
+                internal_syncd_api_send_response(api, status);
```

Comment:
the else case is redundant, 'internal_syncd_api_send_response' is called at the beginning of the previous 'else' layer.